samples/unixdomainsockets/src/main/java/okhttp3/unixdomainsockets/ClientAndServer.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.unixdomainsockets;
+
+import java.io.File;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+
+/**
+ * Create UNIX domain sockets for MockWebServer and OkHttp and connect 'em together. Note that we
+ * cannot do TLS over domain sockets.
+ */
+public class ClientAndServer {
```

Comment:
Server completes noisily

```
WARNING: MockWebServer[1] failed unexpectedly
java.io.IOException: accept failed: Software caused connection abort
	at jnr.unixsocket.UnixServerSocketChannel.accept(UnixServerSocketChannel.java:61)
	at okhttp3.unixdomainsockets.UnixDomainServerSocketFactory$UnixDomainServerSocket.accept(UnixDomainServerSocketFactory.java:85)
	at okhttp3.mockwebserver.MockWebServer$2.acceptConnections(MockWebServer.java:359)
	at okhttp3.mockwebserver.MockWebServer$2.execute(MockWebServer.java:336)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
```

Reply:
Yeah, this might be something we can fix upstream.

Reply:
https://github.com/square/okhttp/issues/4106